llama_hub/lilac_reader/base.py

```
@@ -0,0 +1,100 @@
+"""Lilac reader that loads enriched and labeled Lilac datasets into GPTIndex and LangChain."""
+from typing import List, Optional
+from typing import TYPE_CHECKING, Optional
+
+from llama_index.readers.base import BaseReader
+from llama_index.readers.schema.base import Document
+
+if TYPE_CHECKING:
+    from lilac import FilterLike, Path, ColumnId
+
+class LilacReader(BaseReader):
+    """
+    Lilac dataset reader
+    """
+
+    def load_data(
+        self,
+        dataset: str,
+        text_path: "Path" = 'text',
+        doc_id_path: Optional["Path"] = 'doc_id',
+        columns: Optional[List['ColumnId']] = None,
+        filters: Optional[list["FilterLike"]] = None,
+        project_dir: Optional[str] = None,
+    ) -> List[Document]:
+        """
+        Load text from relevant posts and top-level comments in subreddit(s), given keyword(s) for search
+
+        Args:
+            project_dir (Optional[str]): The Lilac project dir to read from. If not defined, uses the `LILAC_PROJECT_DIR`
+              environment variable.
+            text_path: The path to the text field in the dataset. If not defined, uses 'text'.
+            columns (Optional[List[ColumnId]]): The columns to load from the dataset. If not defined, loads all columns.
+            dataset (str): The dataset to load. Should be formatted like {namespace}/{dataset_name}.
+            filters (Optional[Filter]): A filter to apply to the dataset before loading into documents. Useful to filter
+              for labeled data.
+
+        """
+
+        import lilac as ll
```

can you raise an ImportError if the package is not found?